lib/simple_tsify.js

```diff
@@ -32,7 +34,7 @@ module.exports = function (b, opts) {
         this.push(ts.transpileModule(text, {
           compilerOptions: {
             esModuleInterop: true,
-            jsx: 'react',
+            jsx: ext === '.ts' ? undefined : 'react',
```

Comment:
This makes the default `react` and only turns it off if the extension is `.ts`. I think it should be flipped, so it's only set to `react` if the extension is `.tsx` (maybe `.jsx` too? not sure if that makes sense or is necessary). Seems more future-proof, so if some framework `foo` creates a TypeScript-compatible extension `.tsf`, it doesn't switch to `react` mode.

Reply:
'react' isn't the only option, but probably the most common. Should there be a way to configure this?

Reply:
@chrisbreiding 

Then, how about changing it to `ext === '.js' || ext === '.jsx' || ext === '.tsx'`? Many developers use JSX syntax with `.js` files. I think it's safer to include `.js` files to the list. 

@NicholasBoll

There are 4 options: `undefined`, `"react"`, `"preserve"`, `"react-native"`. 

But `"preserve"` and `"react-native"` don't make sense here. Cypress transpiles the test code and sends it to the browser and runs it. When we select `"preserve"` or `"react-native"` option, the code cannot be executed on browsers. Because JSX is not compiled and they're still `<X />` in the transpiled code.

Currently, developers use `.js`, `.jsx`, `.tsx` extensions to include JSX syntax. And `.ts` doesn't allow JSX syntax. AFAIK, there is no other scenario. So, I don't think providing option to allow other `jsx` TypeScript compile options is an over-spec.

Reply:
> Then, how about changing it to ext === '.js' || ext === '.jsx' || ext === '.tsx'

That sounds reasonable.